src/client/toc.ts

```diff
@@ -13,7 +12,7 @@ if (toc) {
     // If there’s a location.hash, highlight that if it’s at the top of the viewport.
     if (location.hash) {
       for (const heading of headings) {
-        const hash = heading.querySelector<HTMLAnchorElement>("a[href]")?.hash;
+        const hash = `#${heading.id}`;
```

Comment:
I wonder if this needs any encoding? 🤔

Reply:
I think so — since we compare it with location.hash we should probably use

```suggestion
        const hash =  new URL(`#${heading.id}`, location).hash;
```

Reply:
Added `encodeURI`. 👍